cmd/object-handlers_test.go

```diff
@@ -174,6 +175,200 @@ func testAPIGetOjectHandler(obj ObjectLayer, instanceType string, t TestErrHandl
 	}
 }
 
+// Wrapper for calling PutObject API handler tests for both XL multiple disks and FS single drive setup.
+func TestAPIPutObjectStreamSigV4Handler(t *testing.T) {
+	ExecObjectLayerTest(t, testAPIPutObjectStreamSigV4Handler)
+}
+
+func testAPIPutObjectStreamSigV4Handler(obj ObjectLayer, instanceType string, t TestErrHandler) {
+	// get random bucket name.
+	bucketName := getRandomBucketName()
+	objectName := "test-object"
+	// Create bucket.
+	err := obj.MakeBucket(bucketName)
+	if err != nil {
+		// failed to create newbucket, abort.
+		t.Fatalf("%s : %s", instanceType, err)
+	}
+	// Register the API end points with XL/FS object layer.
+	// Registering only the GetObject handler.
+	apiRouter := initTestAPIEndPoints(obj, []string{"PutObject"})
+	// initialize the server and obtain the credentials and root.
+	// credentials are necessary to sign the HTTP request.
+	rootPath, err := newTestConfig("us-east-1")
+	if err != nil {
+		t.Fatalf("Init Test config failed")
+	}
+	// remove the root folder after the test ends.
+	defer removeAll(rootPath)
+
+	credentials := serverConfig.GetCredential()
+
+	bytesDataLen := 1000
+	bytesData := generateBytesData(bytesDataLen)
+	// byte data for PutObject.
+	// test cases with inputs and expected result for GetObject.
+	testCases := []struct {
+		bucketName string
+		objectName string
+		data       []byte
+		dataLen    int
+		// expected output.
+		expectedContent    []byte // expected response body.
+		expectedRespStatus int    // expected response status body.
+	}{
+		// Test case - 1.
+		// Fetching the entire object and validating its contents.
+		{
+			bucketName:         bucketName,
+			objectName:         objectName,
+			data:               bytesData,
+			dataLen:            bytesDataLen,
+			expectedContent:    []byte{},
+			expectedRespStatus: http.StatusOK,
+		},
+	}
+	// Iterating over the cases, fetching the object validating the response.
+	for i, testCase := range testCases {
+		var stream []byte
+		stream = append(stream, []byte("1000;chunk-signature=abcd\r\n")...)
```

Comment:
looks like I can put anything here in chunk-signature, the upload was successful though